its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java

```diff
@@ -1238,28 +1234,23 @@ private void validateCSharpSdk(String folderName) throws IOException {
     runBeginBuildAndEndForStandardProject(folderName, "", true, false);
 
     List<Issue> issues = TestUtils.allIssues(ORCHESTRATOR);
-    if (isTestProjectSupported()) {
-      assertThat(issues).hasSize(3)
-        .extracting(Issue::getRule, Issue::getComponent)
-        .containsExactlyInAnyOrder(
-          tuple(SONAR_RULES_PREFIX + "S1134", folderName + ":Main/Common.cs"),
-          tuple(SONAR_RULES_PREFIX + "S2699", folderName + ":UTs/CommonTest.cs"),
-          tuple(SONAR_RULES_PREFIX + "S2094", folderName + ":Main/Common.cs"));
-      // The AspNetCoreMvc/Views/Home/Index.cshtml contains an external CS0219 issue
-      // which is currently not imported due to the fact that the generated code Index.cshtml.g.cs is in the object folder.
-    } else {
-      assertThat(issues).hasSize(3)
-        .extracting(Issue::getRule, Issue::getComponent)
-        .containsExactlyInAnyOrder(
-          tuple(SONAR_RULES_PREFIX + "S1134", folderName + ":AspNetCoreMvc/Program.cs"),
-          tuple(SONAR_RULES_PREFIX + "S1134", folderName + ":Main/Common.cs"),
-          tuple(SONAR_RULES_PREFIX + "S2094", folderName + ":Main/Common.cs"));
-      // The AspNetCoreMvc/Views/Home/Index.cshtml contains an external CS0219 issue
-      // which is currently not imported due to the fact that the generated code Index.cshtml.g.cs is in the object folder.
-    }
+
+    assertThat(issues).hasSize(2)
+      .extracting(Issue::getRule, Issue::getComponent)
+      .containsExactlyInAnyOrder(
+        tuple(SONAR_RULES_PREFIX + "S1134", folderName + ":Main/Common.cs"),
+        tuple(SONAR_RULES_PREFIX + "S2699", folderName + ":UTs/CommonTest.cs"));
+    // The AspNetCoreMvc/Views/Home/Index.cshtml contains an external CS0219 issue
+    // which is currently not imported due to the fact that the generated code Index.cshtml.g.cs is in the object folder.
   }
 
   private void assertUIWarnings(BuildResult buildResult) {
+    // AnalysisWarningsSensor was implemented starting from analyzer version 8.39.0.47922 (https://github.com/SonarSource/sonar-dotnet-enterprise/commit/39baabb01799aa1945ac5c80d150f173e6ada45f)
+    var analyzerVersion = TestUtils.GetAnalyzerVersion(ORCHESTRATOR);
+    if (analyzerVersion.matches("\\d+\\.\\d+\\.\\d+\\.\\d+")
```

Comment:
DEV and LATEST will throw exception with `Version.create`

Reply:
We do a very similar check here: https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/Tests.java#L75

Would you mind extracting a method and using it in both places?

My personal preference is to use the version with string comparison since it's easier to read.